test/tests/functional/pbs_hook_execjob_end.py

```diff
@@ -286,38 +285,45 @@ def test_execjob_end_reject_request(self):
         self.momB.log_match("Job;%s;JOIN_JOB as node" % jid, n=100,
                             max_attempts=10, interval=2)
 
-        # When the job run approx 10-20 sec, bring sister mom down
-        self.server.expect(JOB, {'job_state': 'R'}, id=jid, offset=20)
+        # When the job run approx 10 sec, bring sister mom down
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid, offset=10)
+        search_after = int(time.time())
         msg = 'mom is not down'
         self.assertTrue(self.momB.stop(), msg)
 
         # Verify momB is down and job is running
+        a = {'state': (MATCH_RE, "down")}
         self.server.expect(
-            NODE, {'state': "down,job-exclusive"}, id=self.momB.shortname)
+            NODE, a, id=self.momB.shortname)
         self.server.expect(JOB, {'job_state': "R"}, id=jid)
 
         # Wait for job to be in E state
-        self.server.expect(JOB, {'job_state': "E"}, id=jid, offset=30)
+        time.sleep(30)
+        self.server.expect(JOB, {'job_state': "E"}, id=jid, offset=20)
 
         # Following message should be logged on momA after job delete request
         # received
         msg = "%s;Unable to send delete job request to one or more" % jid
         msg += " sisters"
 
-        self.momA.log_match(msg, interval=2)
+        self.momA.log_match(msg, interval=2, starttime=search_after)
 
         # Following message should be logged on momA while execjob_end hook is
         # in sleep
+        hook_execution_time = int(time.time())
         self.momA.log_match("Job;%s;executed execjob_end hook" % jid,
-                            n=100, max_attempts=10, interval=2)
+                            starttime=search_after, max_attempts=10,
```

Comment:
starttime for above should be time before job goes to E state since end hook will be run when job ends.